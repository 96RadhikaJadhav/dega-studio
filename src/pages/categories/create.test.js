import React from 'react';
import { BrowserRouter as Router, useHistory } from 'react-router-dom';
import renderer, { act } from 'react-test-renderer';
import { useDispatch, Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import '../../matchMedia.mock';
import CreateCategory from './create';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: jest.fn(),
}));

jest.mock('../../actions/categories', () => ({
  getCategories: jest.fn(),
  addCategory: jest.fn(),
}));

describe('Categories List component', () => {
  let store;
  let mockedDispatch;

  beforeEach(() => {
    store = mockStore({
      categories: {
        req: [],
        details: {},
        loading: true,
      },
      media: {
        req: [],
        details: {},
        loading: true,
      },
    });
    store.dispatch = jest.fn(() => ({}));
    mockedDispatch = jest.fn();
    useDispatch.mockReturnValue(mockedDispatch);
  });
  it('should render the component', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <CreateCategory />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
