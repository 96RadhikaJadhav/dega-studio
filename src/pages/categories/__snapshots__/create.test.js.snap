// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories List component should render the component 1`] = `
<form
  className="ant-form ant-form-horizontal"
  id="create-category"
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-10 ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        htmlFor="create-category_parent_id"
        title="Parent Category"
      >
        Parent Category
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <div
            className="ant-select ant-select-single ant-select-show-arrow ant-select-loading"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
          >
            <div
              className="ant-select-selector"
              onMouseDown={[Function]}
            >
              <span
                className="ant-select-selection-search"
              >
                <input
                  aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                  aria-autocomplete="list"
                  aria-controls="rc_select_TEST_OR_SSR_list"
                  aria-haspopup="listbox"
                  aria-owns="rc_select_TEST_OR_SSR_list"
                  autoComplete="off"
                  className="ant-select-selection-search-input"
                  id="rc_select_TEST_OR_SSR"
                  onChange={[Function]}
                  onCompositionEnd={[Function]}
                  onCompositionStart={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  onPaste={[Function]}
                  readOnly={true}
                  role="combobox"
                  style={
                    Object {
                      "opacity": 0,
                    }
                  }
                  unselectable="on"
                  value=""
                />
              </span>
              <span
                className="ant-select-selection-item"
              />
            </div>
            <span
              aria-hidden={true}
              className="ant-select-arrow ant-select-arrow-loading"
              onMouseDown={[Function]}
              style={
                Object {
                  "WebkitUserSelect": "none",
                  "userSelect": "none",
                }
              }
              unselectable="on"
            >
              <span
                aria-label="loading"
                className="anticon anticon-loading"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  className="anticon-spin"
                  data-icon="loading"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="0 0 1024 1024"
                  width="1em"
                >
                  <path
                    d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
                  />
                </svg>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-10 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="create-category_name"
        title="Category Name"
      >
        Category Name
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="create-category_name"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-10 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="create-category_slug"
        title="Slug"
      >
        Slug
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="create-category_slug"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-10 ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        htmlFor="create-category_description"
        title="Description"
      >
        Description
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <textarea
            className="ant-input"
            id="create-category_description"
            onChange={[Function]}
            onKeyDown={[Function]}
            style={Object {}}
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-10 ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        htmlFor="create-category_medium_id"
        title="Upload Media"
      >
        Upload Media
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <div
            className="ant-space ant-space-vertical"
          >
            <div
              className="ant-space-item"
              style={Object {}}
            >
              <button
                className="ant-btn"
                onClick={[Function]}
                type="button"
              >
                <span>
                  Select
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-14 ant-col-offset-10 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <div
            className="ant-space ant-space-horizontal ant-space-align-center"
          >
            <div
              className="ant-space-item"
              style={
                Object {
                  "marginRight": 8,
                }
              }
            >
              <button
                className="ant-btn ant-btn-primary"
                onClick={[Function]}
                type="submit"
              >
                <span>
                  Submit
                </span>
              </button>
            </div>
            <div
              className="ant-space-item"
              style={Object {}}
            >
              <button
                className="ant-btn"
                onClick={[Function]}
                type="button"
              >
                <span>
                  Reset
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;
